doc/copp/CoPP Config and Management.md

```diff
@@ -0,0 +1,336 @@
+
+### 00-copp.config.json
+This file shall be modified to be compatible to Config DB schema, currently placed at ```swssconfig/sample/00-copp.config.json```. It shall be renamed to ```copp_cfg.j2``` and moved to ```files/image_config/copp/copp_cfg.j2```
+
+### copp_cfg.json
+A new file ```copp_cfg.json``` shall be introduced to include default CoPP tables from copp_cfg.j2 placed under ```/etc/sonic```. This file shall be read by ```coppmgr``` during initialization. 
+
+### Default CoPP Tables
+There are two proposals for loading the default CoPP Tables.
+
+1. Add default CoPP Tables to be part of init_cfg.json. As part of the post start action, the Config DB shall be loaded with default CoPP tables and/or any previously configured CoPP tables as part of the following handler in ```files/build_templates/docker_image_ctl.j2```
+
+```
+    function postStartAction()
+        ...
+        if [ -r /etc/sonic/config_db.json ]; then
+            sonic-cfggen -j /etc/sonic/config_db.json --write-to-db
+            if [ -r /etc/sonic/init_cfg.json ]; then
+                sonic-cfggen -j /etc/sonic/init_cfg.json -j /etc/sonic/config_db.json --write-to-db
+            else
+                sonic-cfggen -j /etc/sonic/config_db.json --write-to-db
+            fi
+        fi
+```
+However, this approach has the following limitations:
+
+In warmboot scenarios, as init_cfg.json is not read during warmboot and may result in ***new*** CoPP Tables in the "to" image not getting applied.
+In coldboot, if user saves the config and the new release happened to have new CoPP values, it will not be applied since previous default values are present in config_db
+
+2. ```coppmgr``` reading directly from ```copp_cfg.json``` and apply the default CoPP Tables. 
+
+For this design, considering the limitations, proposal is to go ahead with second option. 
+
+## Warmboot and Backward Compatibility
+It is desirable to have warmboot functionality from previous release versions of Sonic. Since the existing schema has COPP Group name/key with protocol names (e.g `"COPP_TABLE:trap.group.bgp.lacp"`, there is a limitation in adding any new protocol or trap to an existing CoPP group and seamlessly migrate. However, with this proposal, the following options are considered:
+
+1. The implementation is to do a migration of APP DB entries to new schema.
+2. Let ```db_migrator``` remove all previously saved APP_DB entries and let syncd reconcile and remove the TRAP entries during warmboot. Later, ```coppmgr``` can reapply the traps afresh. Also remove COPP_TABLES from being saved in [backup_database](https://github.com/Azure/sonic-utilities/blob/master/scripts/fast-reboot#L234). This may require additional tests to confirm system behaviour during warmboot. 
+3. Remove '00-copp.config.json' from being included for checksum calculation in ```files/build_scripts/generate_asic_config_checksum.py```
```

Comment:
@prsunny even though we remove copp.json from checksum calculation it will still throw error on warm-reboot from old image.

Eg: current_image let's say 201911 next_image 202011 (that have new copp design)

when we issue warm-boot on 201911 image to go to 202011 asic checksum validation done by 201911 will fail as it will try to get checksum of 2020 image but it will not be same as 201911.

Reply:
Agree, but I think, would recommend to use "warm-reboot-f".